### PR TITLE
Update to 0.23 kafka - 4th try

### PR DESCRIPTION
**Pull request by aliok**
Based on https://github.com/openshift-knative/serverless-operator/pull/1125

Add some logic to keep configMapHash annotation key on `kafka-ch-dispatcher`.

### CODE REVIEW

**Comment by matzew** [comment]
@aliok cool!

**Comment by matzew** [comment]
testing all again ...

**Comment by matzew** [comment]
@aliok can you revert this commit ? https://github.com/openshift-knative/serverless-operator/pull/1126/commits/bd753ae0f89a2a740fe7bdc3e46450b23d7ddc87

since we actually want that - now that we know the pod re spin was caused due to the new annotation on the deplouments

File: knative-operator/pkg/controller/knativekafka/configmaphashtransform.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const configMapHashAnnotationKey = "kafka.eventing.knative.dev/configmap-hash"
+
+// configMapHashTransform makes kafka-ch-dispatcher keep its current configmap hash annotation set by controller
+func configMapHashTransform(client mf.Client) mf.Transformer {
```
Comment (matzew) [inline knative-operator/pkg/controller/knativekafka/configmaphashtransform.go:15]: Perhaps in a follow up PR, we can move the transformations to a sub folder, and perhaps see if the can share some of the duplicated code snippets here and in the `replica transform` logic

but that's fine to be done at a later point

**Comment by aliok** [comment]
bd753ae is now reverted.
Unit tests are added.

**Comment by matzew** [comment]
/test 4.7-upgrade-tests-aws-ocp-47

Let's see if that test passes again

**Comment by matzew** [comment]
/lgtm
/approve

/hold  
(untl the upgrade is green again)

**Comment by matzew** [comment]
/unhold

was green again (upgrade test)

**Pull request merged**